web/preferences.js

```diff
@@ -44,8 +60,12 @@ var Preferences = {
    *                   have been initialized.
    */
   initialize: function preferencesInitialize() {
-    return this.initializedPromise =
-        this._readFromStorage(DEFAULT_PREFERENCES).then(function(prefObj) {
+    return this.initializedPromise = defaultPreferences.then(
+        function (defaults) {
+      this.defaults = defaults;
```

Comment:
Since the `defaults` are now exposed on `Preferences`, can we use `Object.freeze(defaults)` here to prevent outside tampering with the default values?